### DISCUSSION TITLE
Enhancement Idea - add CSV Import config setting to report all errors

### DISCUSSION THREAD

**Discussion by chrisfw** [duckdb/duckdb#4195] [General]
Hi,
Thanks for all of your great work on duckdb, it is an excellent multifaceted tool and database!
 
I just wanted to make an enhancement suggestion.  It would definitely be beneficial to be able to invoke the csv reader with a configuration setting that reported all errors rather than stopping on the first one.  This would facilitate a better user experience as in my example case there were a significant number of the errors like below and they had to be identified by running the same statement over and over until each occurrence had been manually corrected in the source CSV file. 

Thanks,
Chris Whelan

```
D create or replace table customer as select nextval('customerseq') as rownumber, c.* from read_csv_auto('E:\data\tmw\net_new_staging\groupA\STAGE_CUST_INFO.csv',header=True,all_varchar=True,filename=True) c;
Error: Invalid Input Error: Error in file "E:\data\tmw\net_new_staging\groupA\STAGE_CUST_INFO.csv" on line 9556: quote should be followed by end of value, end of row or another quote. (DELIMITER=',' (auto detected), QUOTE='"' (auto detected), ESCAPE='' (auto detected), HEADER=1, SAMPLE_SIZE=10240, IGNORE_ERRORS=0, ALL_VARCHAR=1)
```

**Comment by Alex-Monahan** [comment]
Howdy!

Have you tried using the ignore_errors option?

#3455

**Comment by chrisfw** [comment]
Hi, thanks for your quick reply, but I couldn't get the ignore_errors option to be recognized even though it is displayed in the error output as an option that is set to false.

```
D create or replace table customer as select nextval('customerseq') as rownumber, c.* from read_csv_auto('E:\data\tmw\net_new_staging\groupA\STAGE_CUST_INFO.csv',header=True,all_varchar=True,filename=True,ignore_errors=True) c;
Error: Binder Error: Invalid named parameter "ignore_errors" for function read_csv_auto
Candidates:     quote VARCHAR
    sep VARCHAR
    sample_chunk_size BIGINT
    auto_detect BOOLEAN
    columns ANY
    maximum_line_size VARCHAR
    delim VARCHAR
    escape VARCHAR
    nullstr VARCHAR
    header BOOLEAN
    sample_size BIGINT
    sample_chunks BIGINT
    skip BIGINT
    all_varchar BOOLEAN
    dateformat VARCHAR
    timestampformat VARCHAR
    normalize_names BOOLEAN
    compression VARCHAR
    filename BOOLEAN
    max_line_size VARCHAR
```

**Comment by Alex-Monahan** [comment]
Would you mind adding the other details that the issue template asks for? That will help us understand the issue with the current ignore_errors functionality!

Thanks!

**Comment by chrisfw** [comment]
---
name: Enhancement Request ->  add CSV Import config setting to report all errors.  _(Using ignore_errors which was a suggested means of achieving the same goal currently doesn't appear to work in the CLI)_
---

#### What happens?
```
D create or replace table customer as select nextval('customerseq') as rownumber, c.* from read_csv_auto('E:\data\tmw\net_new_staging\groupA\STAGE_CUST_INFO.csv',header=True,all_varchar=True,filename=True,ignore_errors=True) c;
Error: Binder Error: Invalid named parameter "ignore_errors" for function read_csv_auto
Candidates:     quote VARCHAR
    sep VARCHAR
    sample_chunk_size BIGINT
    auto_detect BOOLEAN
    columns ANY
    maximum_line_size VARCHAR
    delim VARCHAR
    escape VARCHAR
    nullstr VARCHAR
    header BOOLEAN
    sample_size BIGINT
    sample_chunks BIGINT
    skip BIGINT
    all_varchar BOOLEAN
    dateformat VARCHAR
    timestampformat VARCHAR
    normalize_names BOOLEAN
    compression VARCHAR
    filename BOOLEAN
    max_line_size VARCHAR
```
#### To Reproduce
Steps to reproduce the behavior. 
*It is probably not necessary to create a sequence or create a table from the CSV select as displayed in the above.  Attempt to use ignore_errors=True as a CSV option

#### Environment (please complete the following information):
 - OS: Windows
 - DuckDB Version: v0.4.0  (da9ee490d)
 - DuckDB Client: CLI

#### Identity Disclosure:
 - Full Name: Chris Whelan
 - Affiliation: Systems and Software

If the above is not given and is not obvious from your GitHub profile page, we might close your issue without further review. Please refer to the [reasoning behind this rule](https://berthub.eu/articles/posts/anonymous-help/) if you have questions.


#### Before Submitting

- [] **Have you tried this on the latest `master` branch?**
**_I am using the CLI_**
* **Python**: `pip install duckdb --upgrade --pre`
* **R**: `install.packages("https://github.com/duckdb/duckdb/releases/download/master-builds/duckdb_r_src.tar.gz", repos = NULL)`
* **Other Platforms**: You can find binaries [here](https://github.com/duckdb/duckdb/releases/tag/master-builds) or compile from source.

- [ ] **Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?**

**Comment by chrisfw** [comment]
@Alex-Monahan , is the above format and content ok?
Thanks,
Chris
